### PR TITLE
Remove mockito deps from dev

### PR DESCRIPTION
**Pull request by jonahwilliams**
Follow up from https://github.com/flutter/flutter/pull/77748 . Need to rebase after that lands

Remove all usage of mockito in dev.

### CODE REVIEW

File: dev/bots/test/prepare_package_test.dart [hunk 1]
```diff
@@ -129,17 +136,9 @@ void main() {
           if (platform.isWindows) '7za a -tzip -mx=9 $archiveName flutter': null
           else if (platform.isMacOS) 'zip -r -9 --symlinks $archiveName flutter': null
           else if (platform.isLinux) 'tar cJf $archiveName flutter': null,
-        };
+        }));
         await creator.initializeRepo();
         await creator.createArchive();
-        expect(
```
Comment (jonahwilliams) [inline dev/bots/test/prepare_package_test.dart:135]: This is a sacrifice I'm willing to make...

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by christopherfujino** [comment]
RIP mockito

**Comment by christopherfujino** [comment]
Linux tool_integration_tests is failing with wha tlooks like an unrelated flake? https://ci.chromium.org/p/flutter/builders/try/Linux%20SDK%20Drone/234286?

```
02:15 +1 -1: test/integration.shard/deferred_components_test.dart: simple build appbundle android-arm64 target succeeds [E]                                                                            
  Expected: contains 'app-release.aab'
    Actual: 'Downloading android-arm-profile/linux-x64 tools...                 536ms\n'
              'Downloading android-arm-release/linux-x64 tools...                 220ms\n'
              'Downloading android-arm64-profile/linux-x64 tools...               189ms\n'
              'Downloading android-arm64-release/linux-x64 tools...               145ms\n'
              'Downloading android-x64-profile/linux-x64 tools...                 138ms\n'
              'Downloading android-x64-release/linux-x64 tools...                 116ms\n'
              'Deferred components prebuild validation passed.\n'
...
```

**Comment by christopherfujino** [comment]
Looks like Gary addressed this upstream: https://github.com/flutter/flutter/commit/4d38ca6d468c873aa74c6c898c32ee054fd7e2a4

**Pull request merged**